src/utils/transaction.ts

```diff
@@ -20,35 +22,121 @@ import {
   getDecimalFromEtherUnit,
   gasPriceToBase
 } from '@services/EthService';
-import { ITxReceipt, ExtendedAsset, Network, ITxConfig, StoreAccount, TAddress } from '@types';
+import {
+  ExtendedAsset,
+  Network,
+  ITxConfig,
+  StoreAccount,
+  TAddress,
+  ITxType,
+  ITxStatus,
+  IPendingTxReceipt,
+  ITxHash,
+  IFailedTxReceipt,
+  ISuccessfulTxReceipt,
+  ITxReceipt,
+  ITxHistoryStatus
+} from '@types';
+
+export const constructTxReceiptFromTransactionResponse = (txResponse: TransactionResponse) => (
+  txType: ITxType,
+  txConfig: ITxConfig,
+  assets: ExtendedAsset[],
+  txHash?: ITxHash
+) => (status: ITxStatus.PENDING): IPendingTxReceipt => {
+  const useTxResponse = txResponse && !isEmpty(txResponse) && txResponse !== null;
+
+  const to = useTxResponse ? txResponse.to : txConfig.rawTransaction.to;
+  const data = useTxResponse ? txResponse.data : txConfig.rawTransaction.data;
+  const value = useTxResponse ? txResponse.value : bigNumberify(txConfig.rawTransaction.value);
+
+  const contractAsset = getAssetByContractAndNetwork(to, txConfig.network)(assets);
+  const baseAsset = getBaseAssetByNetwork({ network: txConfig.network, assets });
+
+  return {
+    asset: contractAsset || txConfig.asset,
+    baseAsset: baseAsset || txConfig.baseAsset,
```

Comment:
Because we're fetching data from a list of assets, even when it /should/ return an asset, sometimes it just doesn't. which results in the whitescreen bugs we have every few weeks (like the baseAsset bug we had 2 weeks ago). This just sort of handles that if it doesn't exist.

Reply:
Will be solved by using just the txConfig.